docs/extras/use_cases/web_scraping/web_scraping_with_openai_functions.ipynb

```diff
@@ -0,0 +1,389 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "6605e7f7",
+   "metadata": {},
+   "source": [
+    "# Web scraping using OpenAI Functions Extraction chain\n",
+    "\n",
+    "Web scraping is challenging for many reasons; one of them is the changing nature of modern websites' layouts and content, which requires modifying scraping scripts to accommodate the changes.\n",
+    "\n",
+    "Using OpenAI Function in the extraction chain, you can avoid having to change your code constantly when websites change. Plus, this will save you time setting up when scraping a website for the first time by creating simple schemas.\n",
+    "\n",
+    "In this notebook, we scrape The Wall Street Journal. You'll need to have an OpenAI API key for this example."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "05087ad3",
+   "metadata": {},
+   "source": [
+    "## Install dependencies\n",
+    "\n",
+    "We need OpenAI and LangChain for the extraction part, and Playwright and Beautiful Soup for the scraping part."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "00c96747",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n",
+      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip is available: \u001b[0m\u001b[31;49m23.0.1\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m23.2.1\u001b[0m\n",
+      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49mpython3.11 -m pip install --upgrade pip\u001b[0m\n",
+      "Note: you may need to restart the kernel to use updated packages.\n"
+     ]
+    }
+   ],
+   "source": [
+    "pip install -q openai langchain playwright beautifulsoup4"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5ef7f514",
+   "metadata": {},
+   "source": [
+    "## Create a simple scraper function\n",
```

Comment:
OK, this is cool.
 
1/ Let's create a new web loader for Chromium that wraps this logic.

2/ It can follow what we did here:

https://github.com/langchain-ai/langchain/pull/8036

In particular, see:

https://github.com/rlancemartin/langchain/blob/b8a1c28a88917ba4561d53f970dc2df26a9da73a/langchain/document_loaders/async_html.py

I ran what you have and compared to this:

```
loader = AsyncHtmlLoader(url)
docs = loader.load()
html2text = Html2TextTransformer()
docs = html2text.transform_documents(docs)
html_content=docs[0].page_content
```

(See docs [here](https://python.langchain.com/docs/integrations/document_transformers/html2text)).

I found found that Chromium is indeed better. 

For some reason, `html2text` is loosing the news article summaries. 

So, we should add it as a web loader and simply import here.

Reply:
Nice. Will change mine

Reply:
Just took care of this for you!

Reply:
woo hoo